scripts/symlink_home

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python3
+import argparse
+import os
+import sys
+from os import path
+from typing import List
+
+
+def main(argv: List[str]) -> int:
+    parser = argparse.ArgumentParser(description="symlink n files to a base path")
+    parser.add_argument("files", metavar="file", nargs="+", help="file to symlink")
+    parser.add_argument("--base_src", required=True, help="base src directory")
+    parser.add_argument("--base_dest", required=True, help="base dest directory")
+    args = parser.parse_args(argv[1:])
```

Comment:
neat trick, if you do this:

```python
def main(argv: Optional[Sequence[str]] = None) -> int:
    ...
    args = parser.parse_args(argv)


if __name__ == '__main__':
    exit(main())
```

this does the right thing (argparse will by default look at `sys.argv[1:]`) and it allows you to test it (if you wanted to write tests) via `main(('a', 'b', 'c'))` or `main(['d', 'e', '--f'])`

Reply:
oh good point. originally i was consuming argv[0] but i moved that out of the python bits.